src/test/FreeC/Pass/ExportPassTests.hs

```diff
@@ -0,0 +1,117 @@
+-- | This module contains tests for "FreeC.Pass.ExportPass".
+
+module FreeC.Pass.ExportPassTests
+  ( testExportPass
+  )
+where
+
+import           Test.Hspec
+
+import           Data.List                      ( find )
+import           Data.Maybe                     ( fromJust )
+import qualified Data.Set                      as Set
+import qualified Data.Text                     as Text
+
+import qualified FreeC.Backend.Coq.Syntax      as Coq
+import           FreeC.Environment
+import           FreeC.Environment.Entry
+import           FreeC.Environment.ModuleInterface
+import           FreeC.IR.Syntax               as IR
+import           FreeC.Monad.Class.Testable
+import           FreeC.Monad.Converter
+import           FreeC.Pass.ExportPass
+import           FreeC.Pretty
+import           FreeC.Test.Environment
+import           FreeC.Test.Parser
+
+-- | looks up an exported entry in the given module interface
+lookupExportedEntry :: IR.Scope -> IR.QName -> ModuleInterface -> Maybe EnvEntry
+lookupExportedEntry scope qname moduleInterface =
+  if (scope, qname) `elem` Set.toList (interfaceExports moduleInterface)
+    then find ((qname ==) . entryName)
+              (Set.toList $ interfaceEntries moduleInterface)
+    else Nothing
+
+-- | checks if the given 'Coq.Qualid' is qualified and compares the qualifier
+--   to the given module name.
+--   The check succeeds if the qualifier and module name are the same.
+shouldBeQualifiedWith :: Coq.Qualid -> String -> Converter Expectation
+shouldBeQualifiedWith qualid modNameStr = do
+  case qualid of
+    Coq.Qualified s _ -> return $ Text.unpack s `shouldBe` modNameStr
+    Coq.Bare _ ->
+      return
+        $    expectationFailure
+        $    showPretty
+        $    prettyString "Expected qualifier"
+        <$$> line
+        <>   indent 2 (prettyString modNameStr)
+        <>   line
+        <$$> prettyString "but it was not qualified"
+
+testExportPass :: SpecWith ()
```

Comment:
We usually don't need `SpecWith`. There is a type synonym `Spec` for `SpecWith ()`.

```suggestion
-- | Test group for 'exportPass' tests.
testExportPass :: Spec
```